File: frontend/lib/data/models/region.dart
import 'package:json_annotation/json_annotation.dart';

part 'region.g.dart';

@JsonSerializable()
class Region {
	final int id;
	final int? parentId;
	final String name;
	final String? nameLocal;
	final int level;
	final String code;
	final double? latitude;
	final double? longitude;
	final String? description;
	final String? imageUrl;
	final int status;

	Region({
		required this.id,
		this.parentId,
		required this.name,
		this.nameLocal,
		required this.level,
		required this.code,
		this.latitude,
		this.longitude,
		this.description,
		this.imageUrl,
		this.status = 1,
	});

	factory Region.fromJson(Map<String, dynamic> json) => _$RegionFromJson(json);
	Map<String, dynamic> toJson() => _$RegionToJson(this);
}

File: backend/src/main/java/com/regional/calendar/entity/Region.java
package com.regional.calendar.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("region")
public class Region {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long parentId;

    private String name;

    private String nameLocal;

    private String namePinyin;

    private Integer level;

    private String code;

    private String zipCode;

    private String areaCode;

    private BigDecimal latitude;

    private BigDecimal longitude;

    private BigDecimal areaSize;

    private Long population;

    private String description;

    private String imageUrl;

    private String gallery;

    private String tags;

    private Integer sortOrder;

    private Integer status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}
